src/main/java/com/pinterest/secor/parser/MessageParser.java

```diff
@@ -48,13 +52,22 @@ public MessageParser(SecorConfig config) {
             String separatorPattern = Pattern.quote(mConfig.getMessageTimestampNameSeparator());
             mNestedFields = mConfig.getMessageTimestampName().split(separatorPattern);
         }
+        try {
+            messageTransformer = ReflectionUtil
```

Comment:
Thinking more about it, this is not the right place for this logic as it has nothing to do with extracting the partitions (that's what the parser does). I would suggest moving it to Consumer.java